### DISCUSSION TITLE
Datahub Importer - HTTP - Data Selector for JSON Payload

### DISCUSSION THREAD

**Discussion by cancan101** [pimcore/pimcore#17670] [Q&A / Support]
Is there any way to control the "JSONPath" of the data payload to be importer when using an HTTP Data Source? For example the endpoint returns the data like this:
```json
{
  "data": {...},
  "pagination": {...}
}
```
and I want to tell Pimcore that the relevant data can be found under the `data` key. Other REST / Json importers generally offer a concept of an extraction "path" that specifies where the data can be found. (something like this: https://jmespath.org/).

**Comment by fashxp** [comment]
Nope, see https://pimcore.com/docs/platform/Data_Importer/Configuration/File_Formats#json

**Reply by cancan101** [reply]
>The adapter expects an array of json objects and reads all first level attributes as separate fields.

Seems to be the limitation I am running into. That prevents me from using the transform pipeline to pull out the fields that I need.

Would Pimcore be open to a PR adding a path extractor for JSON? Similar fields already exist for XML (XPath) and to a lesser extent CSV / Excel ("Skip First Row").

**Reply by fashxp** [reply]
yes, definitely!

**Reply by cancan101** [reply]
PR: https://github.com/pimcore/data-importer/pull/437
